shipyard-controller/handler/projectmanager.go

```diff
@@ -184,6 +185,36 @@ func (pm *ProjectManager) Update(params *models.UpdateProjectParams) (error, com
 		return ErrProjectNotFound, nilRollback
 	}
 
+	if params.Shipyard != nil && *params.Shipyard != "" {
+		shipyard := &keptnv2.Shipyard{}
+		decodedShipyard, _ := base64.StdEncoding.DecodeString(*params.Shipyard)
+		_ = yaml.Unmarshal([]byte(decodedShipyard), shipyard)
+		var expandedStages []*models.ExpandedStage
+
+		for _, s := range shipyard.Spec.Stages {
+			es := &models.ExpandedStage{
+				Services:  []*models.ExpandedService{},
+				StageName: s.Name,
+			}
+			expandedStages = append(expandedStages, es)
+		}
+
+		newProject := &models.ExpandedProject{
+			CreationDate:    strconv.FormatInt(time.Now().UnixNano(), 10),
+			GitRemoteURI:    params.GitRemoteURL,
+			GitUser:         params.GitUser,
+			ProjectName:     *params.Name,
+			Shipyard:        string(decodedShipyard),
+			ShipyardVersion: shipyardVersion,
+			Stages:          expandedStages,
+		}
+
+		err = pm.validateShipyardStagesUnchaged(oldProject, newProject)
+		if err != nil {
+			return ErrInvalidStageChange, nilRollback
+		}
```

Comment:
I would move this code into a function (for example `validateShipyardUpdate(params *models.UpdateProjectParams)`) and call this code later at line 269
This way, we have the logic of handling a change in the shipyard definition in a single place.